mypy/build.py

```diff
@@ -2754,7 +2754,7 @@ def load_graph(sources: List[BuildSource], manager: BuildManager,
 
     # Note: Running this each time could be slow in the daemon. If it's a problem, we
     # can do more work to maintain this incrementally.
-    seen_files = {st.path: st for st in graph.values() if st.path}
+    seen_files = {os.path.abspath(st.path): st for st in graph.values() if st.path}
```

Comment:
This code is very performance critical, and `os.path.abspath` is too slow to call here since it can make a syscall. We'd need a way to do this without introducing much overhead (graph may have 100k+ values).

Reply:
Thank you for your review!
I understand. Then, how about looking up both absolute path and relative path of `newst.path` in `seen_files` at this part:
https://github.com/python/mypy/blob/d9c5c65d3421af67fe8ecc598430f85ad87c8319/mypy/build.py#L2805-L2807

Reply:
I have updated my PR: https://github.com/python/mypy/pull/9059/commits/c1f7d1e4c458cf2bdb57e2ef23e9ed0ca5c6dc24
Could you please review if it is reasonable?

Reply:
How about going back to the original approach, but adding an `abspath` field to `State` that is computed in State's constructor, instead of recomputing it on each load_graph?

(In the daemon, load_graph gets called a lot on a big graph but with only a few sources changed. We do some things that are linear in the graph but we feel bad about it and try to make sure that they aren't *expensive* things.)

Reply:
OK actually I just went and did that thing. Thank you!